### PR TITLE
package.json,CI.yml: add devDependencies

### PR DESCRIPTION
**Pull request by Mersho**
These are packages that are only needed for local development and testing and adding them will make it easier to develop your application. They are not required when your application is deployed in production.

### CODE REVIEW

**Comment by knocte** [comment]
> These are packages that are only needed for local development and testing and adding them will make it easier to develop your application. They are not required when your application is deployed in production.

Define "make it easier to develop".

> Also print the typescript version on the CI as well.

You sure?

**Comment by Mersho** [comment]
> Define "make it easier to develop".

When you have devDependencies, you can conveniently execute the yarn command to install packages. However, without devDependencies, you might encounter a ‘module not found’ error each time you run tests.

**Comment by Mersho** [comment]
Also we use the `yarn add --dev` command in our CI process, which we then transfer to devDependencies.

**Comment by Mersho** [comment]
> commands are not needed by our CI recipe anymore?

No we don't need them anymore. We can simply execute yarn and packages will be installed from packages.json.

File: package.json [hunk 1]
```diff
@@ -1,5 +1,16 @@
 {
     "dependencies": {
-      "cosmiconfig": "8.0.0"
+        "commitlint": "*",
+        "conventional-changelog-conventionalcommits": "*",
+        "cosmiconfig": "8.0.0"
```
Comment (knocte) [inline package.json:5]: @Mersho WHY ARE YOU ADDING THIS NOW?????????? WE WERE ONLY TALKING ABOUT DEV DEPENDENCIES HERE! THE COMMIT MESSAGE ONLY TALKS ABOUT DEV DEPENDENCIES!!!!!!!!!!!!!!!!!!

Comment (knocte) [reply package.json:5]: WHY ARE YOU MIXING AGAIN CHANGES UNRELATED???????? WHY IS THIS PRINCIPLE SO HARD TO UNDERSTAND??????????

File: package.json [hunk 2]
```diff
@@ -1,5 +1,14 @@
 {
     "dependencies": {
-      "cosmiconfig": "8.0.0"
+        "cosmiconfig": "8.0.0"
+    },
+    "devDependencies": {
+        "@commitlint/types": "^17.4.4",
```
Comment (knocte) [inline package.json:6]: @Mersho the packages being removed from the `add --dev` call are: jest typescript ts-jest @types/jest ts-node, NOT THIS ONE!

File: .github/workflows/CI.yml [hunk 3]
```diff
@@ -94,10 +94,10 @@ jobs:
         run: |
           npm install conventional-changelog-conventionalcommits
           npm install commitlint@latest
-      - name: Install yarn
+      - name: Install yarn & packages
         run: |
           npm install --global yarn
-          yarn add --dev jest typescript ts-jest @types/jest
+          yarn
```
[the remark this answers is not in the document]
Comment (Mersho) [reply .github/workflows/CI.yml:100]: > @Mersho why do you leave this here? if we're not using `yarn add` command, what does `yarn` alone achieve?

This is the equivalent of running `yarn install`. It looks at the `package.json` file in the current directory and installs all the dependencies listed there.

[the remark this answers is not in the document]
Comment (knocte) [reply .github/workflows/CI.yml:100]: > This is the equivalent of running yarn install

Then why not use `yarn install` which is more readable?

> Also yarn add is used to add a new package to package.json dependencies.

Why do you bring this up?

[the remark this answers is not in the document]
Comment (Mersho) [reply .github/workflows/CI.yml:100]: > Then why not use 
 
 Okay I'm going to use it...

> Why do you bring this up?

Just pointing out & explaining the differences.

**Pull request closed without merging**